lib/Parse/ParseStmt.cpp

```diff
@@ -601,6 +601,11 @@ ParserResult<Stmt> Parser::parseStmt() {
     LLVM_FALLTHROUGH;
   default:
     diagnose(Tok, tryLoc.isValid() ? diag::expected_expr : diag::expected_stmt);
+    if (Tok.is(tok::at_sign)) {
+      // Recover from erroneously placed attribute.
+      consumeToken(tok::at_sign);
+      consumeIf(tok::identifier);
```

Comment:
Ah, thanks! Can you leave a comment about keeping this in sync with `isStartOfStmt` and vice versa?

Reply:
Sure, https://github.com/apple/swift/pull/23368